iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/reports/tflint/TFLintImporter.java

```diff
@@ -0,0 +1,165 @@
+import org.sonar.api.batch.fs.TextRange;
+import org.sonar.api.batch.rule.Severity;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.issue.NewExternalIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rules.RuleType;
+import org.sonar.iac.common.reports.AbstractJsonReportImporter;
+import org.sonar.iac.common.reports.ReportImporterException;
+import org.sonar.iac.common.warnings.AnalysisWarningsWrapper;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONArray;
+import org.sonarsource.analyzer.commons.internal.json.simple.JSONObject;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.ParseException;
+
+public class TFLintImporter extends AbstractJsonReportImporter {
+
+  private static final String MESSAGE_PREFIX = "TFLint report importing: ";
+  // Matches: `: filename.tf:2,21-29:`
+  private static final Pattern FILENAME_PATTERN = Pattern.compile(":\\s([^*&%:]+):(\\d+),(\\d+)-(\\d+):");
+
+  public TFLintImporter(SensorContext context, AnalysisWarningsWrapper analysisWarnings) {
+    super(context, analysisWarnings, MESSAGE_PREFIX);
+  }
+
+  @Override
+  protected List<JSONArray> parseJson(File reportFile) {
+    JSONObject json = null;
+    try {
+      json = (JSONObject) jsonParser.parse(Files.newBufferedReader(reportFile.toPath()));
+    } catch (IOException e) {
+      String message = String.format("could not read report file %s", reportFile.getPath());
+      logWarning(message);
+    } catch (ParseException e) {
+      String message = String.format("could not parse file as JSON %s", reportFile.getPath());
+      logWarning(message);
+    }
+    if (json == null) {
+      return Collections.emptyList();
+    }
+    JSONArray issuesArray = (JSONArray) json.get("issues");
+    JSONArray errorsArray = (JSONArray) json.get("errors");
+
+    issuesArray.addAll(errorsArray);
+
+    return List.of(issuesArray);
+  }
+
+  @Override
+  protected NewExternalIssue toExternalIssue(JSONObject issueJson) {
+    JSONObject rule = (JSONObject) issueJson.get("rule");
+    if (rule == null) {
+      // it's error, not an issue
+      String severity = (String) issueJson.get("severity");
+
+      NewExternalIssue externalIssue = context.newExternalIssue()
+        .ruleId("tflint.error")
+        .type(RuleType.CODE_SMELL)
+        .engineId("tflint")
+        .severity(severity(severity))
+        .remediationEffortMinutes(0L);
+      externalIssue.at(errorLocation(issueJson, externalIssue));
+      return externalIssue;
+    }
+
+    String ruleId = (String) rule.get("name");
+    String severity = (String) rule.get("severity");
+
+    NewExternalIssue externalIssue = context.newExternalIssue()
+      .ruleId(ruleId)
+      .type(RuleType.CODE_SMELL)
+      .engineId("tflint")
+      .severity(severity(severity))
+      .remediationEffortMinutes(0L);
+    externalIssue.at(issueLocation(issueJson, externalIssue));
+    return externalIssue;
```

Comment:
You could have a common NewExternalIssue builder as only location and rule id differ. So you can reduce the complexity of this method.

Reply:
There is also a difference in location calculation: `errorLocation()` vs. `issueLocation()`. Anyway it can be improved